### ISSUE TITLE
Calendar breaks under Angular 8 served in prod

### ISSUE THREAD

**Issue by szala12** [technikhil314/angular-components#46]
I use the plugin in an Angular 8 application, and on building/serving the application with the ```--prod``` flag, the calendar breaks.

1. Make an Angular 8 application.
2. Add calendar to it in some way.
3. Serve using ```ng serve --prod true```

**Expected behavior**
This is what the calendar looks like without ```--prod```:
![image](https://user-images.githubusercontent.com/15906505/98359697-a1431c80-204e-11eb-96ae-4392c1fc69ae.png)
![image](https://user-images.githubusercontent.com/15906505/98359777-bddf5480-204e-11eb-8b62-eec2f46ba0c5.png)

This is what it looks like with ```--prod```
![image](https://user-images.githubusercontent.com/15906505/98359564-75c03200-204e-11eb-9b53-54c27daa17cd.png)
![image](https://user-images.githubusercontent.com/15906505/98359593-7f499a00-204e-11eb-8110-5d7e9022fe0a.png)
![image](https://user-images.githubusercontent.com/15906505/98359450-44476680-204e-11eb-839c-e503902b91d2.png)

I've been able to consistently replicate the issue over all versions of the library, including the ones that used moment.js instead of dayjs. I've tried different styles for importing:
```javascript
import { Dayjs } from 'dayjs';
import * as dayjs from 'dayjs';
import dayjs from 'dayjs';
```

and I've added ```"allowSyntheticDefaultImports": true``` to my tsconfig.json as well. Nothing helps.

**Comment by technikhil314** [comment]
Hi @szala12 

Thanks for finding out this issue. Can you please provide some example codesandbox also can you please confirm if this happens with ng build too? ng serve is not supposed to be ideally run with `--prod true`

Regarding older versions yes those are for angular-2 only. I doubt if they will work with angular 4 or not. So I dont recommend 1.x versions in angular 6 onwards.

**Comment by szala12** [comment]
My bad, I should've mentioned it. I came across it after running ```ng build``` in prod, that's how it came up in the first place. The ```ng serve --prod true``` is only to help replicate on local. I'm not entirely sure how to replicate a prod issue in sandbox, so if you could help me a little with that, that'd be great.

And yeah, we got the issue with angular-datetimerangepicker@2.0.0 as well. I'm not completely sure here, but it looks like whatever webpack does when we build or serve with ```--prod true```, it breaks the import for your included dayjs library, the one in the node_modules directory inside your package.

**Comment by technikhil314** [comment]
Yeah may be it will be difficult to make production bundle in codesandbox. But condesandbox can be used to sync with your github repo. https://codesandbox.io/s/github. 

If you can create a github repo backed sandbox. It will help me to debug and see it works in non-prod bundles.

**Comment by technikhil314** [comment]
@szala12 I was able to reproduce the problem.

Here are my findings

I guess is angulars build optimizer is removing dayjs extentions assuming they are not used. Seems like dayjs or my package needed to marked as not sideeffectfree

try setting `"buildOptimizer": false` in your angular.json

Here is the difference in bundles sizes 

With build optimizer enabled
<img width="997" alt="Screenshot 2020-11-13 at 9 18 43 PM" src="https://user-images.githubusercontent.com/6815560/99092287-f35be300-25f6-11eb-934c-7bc2d945ddbe.png">

without build optimizer
<img width="996" alt="Screenshot 2020-11-13 at 9 20 32 PM" src="https://user-images.githubusercontent.com/6815560/99092349-079fe000-25f7-11eb-8b11-86d698cce1bb.png">

As you can see the main bundle jumps from 451 to 586. Which is significant difference. I ll investigate more and let you know. Thanks for finding out the issue.

**Comment by szala12** [comment]
@technikhil314 Hey. I am terribly sorry about the delay. As weird as this sounds to hear, I've logged this bug on behalf of a colleague who doesn't happen to be on GitHub and is not keen on joining, so they decided to move ahead with another library. I was planning on replicating it this weekend, with it being a holiday proper.

**Comment by technikhil314** [comment]
No worries. As I guessed correctly I missed to mark my package to have side effects on dayjs since I am using some dayjs extentions. The next version I will release tomorrow will have it fixed. Thanks.

**Comment by technikhil314** [comment]
This is fixed in v2.1.1
